src/controllers/userController.ts

```diff
@@ -2,34 +2,62 @@ import { Request, Response } from 'express';
 import UserService from '../services/userService.js';
 import { userUpdateValidation, userValidation } from '../validations/userValidation.js';
 
-export const getAllUsers = (req: Request, res: Response) => {
-  res.status(200).json(UserService.getAllUsers(req.query));
-};
+class UserController {
+  async getAllUsers(req: Request, res: Response) {
+    try {
+      const getAllUsers = await UserService.getAllUsers(req.query);
+      return res.status(200).json(getAllUsers);
+    } catch (error) {
+      res.status(500).json(error);
+    }
+  }
+
+  async getUser(req: Request, res: Response) {
+    try {
+      const getUser = await UserService.getUser(req.params.id);
+      return res.status(200).json(getUser);
+    } catch (error) {
+      res.status(500).json(error);
+    }
+  }
 
-export const getUser = (req: Request, res: Response) => {
-  res.status(200).json(UserService.getUser(req.params.id));
-};
+  async createUser(req: Request, res: Response) {
+    const { error } = await userValidation(req.body);
 
-export const createUser = (req: Request, res: Response) => {
-  const { error } = userValidation(req.body);
+    if (error) {
+      return res.status(400).json(error.details);
+    }
 
-  if (error) {
-    return res.status(400).json(error.details);
+    try {
+      const createUser = await UserService.createUser(req.body);
+      return res.status(200).json(createUser);
+    } catch (error) {
+      res.status(500).json(error);
+    }
```

Comment:
FYI: In this case it's better to use a single error interceptor (but you need to create custom errors to do this, however it will be covered in future modules)